core/src/syscall/precompiles/blake2b/compress/trace.rs

```diff
@@ -0,0 +1,122 @@
+use crate::cpu::MemoryRecordEnum;
+use crate::runtime::ExecutionRecord;
+use crate::syscall::precompiles::blake2b::compress::columns::NUM_BLAKE2B_COMPRESS_INNER_COLS;
+use crate::syscall::precompiles::blake2b::{Blake2bCompressInnerChip, NUM_MIX_ROUNDS};
+use crate::utils::pad_rows;
+use std::borrow::BorrowMut;
+
+use p3_field::PrimeField;
+use p3_matrix::dense::RowMajorMatrix;
+
+use crate::air::MachineAir;
+
+use super::columns::Blake2bCompressInnerCols;
+use super::{
+    MIX_INDEX, MIX_INPUT_SIZE, MSG_ELE_PER_CALL, NUM_MSG_WORDS_PER_CALL, NUM_STATE_WORDS_PER_CALL,
+    OPERATION_COUNT, SIGMA_PERMUTATIONS, STATE_ELE_PER_CALL,
+};
+
+impl<F: PrimeField> MachineAir<F> for Blake2bCompressInnerChip {
+    fn name(&self) -> String {
+        "Blake2bInnerCompress".to_string()
+    }
+
+    fn generate_trace(
+        &self,
+        input: &ExecutionRecord,
+        output: &mut ExecutionRecord,
+    ) -> RowMajorMatrix<F> {
+        let mut rows = Vec::new();
+
+        let mut new_field_events = Vec::new();
+
+        for i in 0..input.blake2b_compress_inner_events.len() {
```

Comment:
For maximum performance, we generally recommend that you use `par_iter` to iterate over the events here so that trace generation for each row can be done in parallel. @jtguibas @ratankaliani might have a good example for you to take a look at.

Reply:
done! please see https://github.com/succinctlabs/sp1/commit/dae55542cd38371935a43e3c8421badded586abb